### ISSUE TITLE
API Proposal: InvSquareRoot for Vector<Single>

### ISSUE THREAD

**Issue by Auburns** [dotnet/runtime#20858]
Having the ability to calculate the inverse square root is much better for performance than square root in several situations, for example: normalising a vector, since it avoids using division. 

Inverse square root would only have an implementation for singles since there is no equivalent intrinsic for doubles, so this is the proposed method:

`public static Vector<Single> InvSquareRoot(Vector<Single> value)`

Which would map directly to these intrinsics:

SSE2:
`__m128 _mm_rsqrt_ps (__m128 a)`

AVX:
`__m256 _mm256_rsqrt_ps (__m256 a)`

NEON:
`float32x4_t vrsqrteq_f32(float32x4_t a)`

**Comment by tannergooding** [comment]
Going to close this for now. This hasn't gained much traction and likely needs more thought around cross platform implementation (such as how it would function on ARM devices) before it could move forward.

If someone feels strongly about this, please feel free to open a new issue. Our API review process is detailed here with a "good example" being listed under step 1. Ideally the new issue would provide many of the sections given, but would at a minimum provide the base rationale and proposed API surface sections.

For this API in particular, the exposure of the System.Runtime.Intrinsic types and the methods for converting to/from Vector<T> should unblock users who immediately need this functionality.
